fpr/migrations/0007_embedded_default_thumbnail.py

```diff
@@ -0,0 +1,100 @@
+mf30vNzvtGIj3KWDx0n9F2gML13Pfvpa1vmNmf314znLPJBJp9+tLWwNhJRhtwW6jJkNw2l2jfcF
+oCYADcIAPABEA4+sQD01nq9VcUaciWEkcnTGpVBRPSCNpfSmFX2F842gNwWc4hzCACIW8RABH0UA
+xdv3cZl2K2rHuPNW85dpbjLIrexrYhqAmLkscHTGsslRFi+yIZZkj48Ozu6QV6VKRvbMivKm0uBv
+Sq9reDUuNFrVZivsa10WSpLlNjhDVBXjT4dJWxsjLVHFR9kua1qNyKbW5Mg6yY3AxKxKAy5PapAm
++43oLuXlPvEsDR3HbLOsLKUyksm0c5BaXVQx4zxmom+OnWTOVy2UQ6Mx56g8SGAqHDqKW2SxxQfL
+LXOJ86VrugaBtURZrPeYisizHj2TWK1B8Tud9oxEe5SweOk/ou0otzvtGIj3KWDx0n9F2qCqqqoC
+pC7VicpPuS5UEouwvrGkZ5PM5LQt5zByliEq6+7gAcbhAq0BuH0iAAHqAKPVI3a07SPJfc+efFjF
+FZivg9z+MRVdF23RldzvtGIj3KWDx0n9F2tru4Lt+ah9ROf4VnnT9P8AEjGvb8PW4ik0fysErbiC
+EjXNHmZNrwyuMXj8sFzNdTpGqRL0S1AyeRbWNIemWPnl5QQxC7zWe5H7TNH4f3OWPricarbXS3mr
+O8SfgSzS12TDnVSM81nuSe0zR/8AGcr/AImq81nuSe0zR/8AGcr/AImqWn6ItYP0JLFefoOdIzaz
+HjuSZND9uj94D/V8TXfWuB2styQfVk7SAH8PGVvriUaY+3ht4ahNNeoTIef8/wCQ8UyBxkGKQxdH
+o9i79TuKe9O4yaNSNxdXVxkcah1zX5LuhyBMhQpkD/e9Xv7goUODEWxpEj1Ktw8LUnOtnBqSk3Wv
+a+ook6pzWeKfNZH/2Q==\"\"\"
+
+def main(target):
+    with open(target, 'w+b') as f:
+        f.write(DEFAULT_THUMBNAIL.decode('base64'))
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--output-location', required=True)
+    args, _ = parser.parse_known_args()
+
+    sys.exit(main(args.output_location))
+"""
+
+
+def data_migration(apps, schema_editor):
+    """Introduce a new FPCommand "Using default thumbnail".
+
+    The existing command expects to find the default thumbnail file in the
+    Archivematica shared directory, see: https://git.io/vHD7L. This migration
+    attempts to remove that level of coupling by introducing a new version of
+    the command as a Python script where the contents of default.jpg are
+    embedded.
+    """
+    FPCommand = apps.get_model('fpr', 'FPCommand')
+    FPRule = apps.get_model('fpr', 'FPRule')
+
+    # Skip if the existing command is not REPLACE_COMMAND_UUID.
+    rule = FPRule.objects.get(enabled=True, purpose='default_thumbnail')
+    if rule.command_id != REPLACE_COMMAND_UUID:
+        return
+
+    replaces = determine_what_replaces_model_instance(FPCommand, rule.command)
+    replaces.enabled = False
+    replaces.save()
+
+    command = FPCommand.objects.create(
+        description=u'Using default thumbnail',
+        command=REPLACE_COMMAND,
+        script_type=u'pythonScript',
+        output_location=u'%outputDirectory%%postfix%.jpg',
+        command_usage='normalization',
+        # <FPTool: Archivematica Script version 1.0>
+        tool_id=u'efa8474a-8526-48c3-8279-e5a76bdc0995',
+        # <FormatVersion: Image (Raster): JPEG: Generic JPEG>
+        output_format_id=u'ffb55e9a-29f1-4276-a9a5-5ef813229b79',
+        # <FPCommand: Standard verification command (file exists only)>
+        verification_command_id=u'ef3ea000-0c3c-4cae-adc2-aa2a6ccbffce',
+        replaces=replaces,
+        enabled=True,
+    )
+
+    rule.command = command
+    rule.save()
```

Comment:
I think you should, yes.  Instead of setting 'rule.command = command' you should set 'rule.enabled = False', and create a new rule, that has 'replaces = rule'.  

Otherwise, when upgrading an older system, the 'count_attempts' and 'count_okay' properties of the FPRule will be misleading.  Those values were incremented before the Rule got changed to use a new command.